Assets/MixedRealityToolkit/_Core/Utilities/EdgeUtilities.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Utilities
+{
+    public static class EdgeUtilities
+    {
+        /// <summary>
+        /// A value that should be larger than the maximum boundary width.
+        /// </summary>
+        /// <remarks>
+        /// This value is used to ensure that line segments are created 
+        /// that will intersect with a piece of the room boundary.
+        /// </remarks>
+        internal static readonly float maxWidth = 10000f;
+
+        /// <summary>
+        /// A value representing an invalid point.
+        /// </summary>
+        public static readonly Vector2 InvalidPoint = new Vector2(float.NegativeInfinity, float.NegativeInfinity);
+
+        /// <summary>
+        /// Determines if the specified point is within the provided geometry.
+        /// </summary>
+        /// <param name="geometryEdges">The geometry for which we are checking the point.</param>
+        /// <param name="point">The point being checked.</param>
+        /// <returns>
+        /// True if the point falls within the geometry, false otherwise.</returns>
+        public static bool IsInsideBoundary(
+            Edge[] geometryEdges, 
+            Vector2 point)
+        {
+            if (geometryEdges.Length == 0)
+            {
+                return false;
+            }
+
+            // Check if a ray to the right (X+) intersects with an 
+            // odd number of edges (inside) or an even number of edges (outside)
+            var rightEdge = new Edge(point, new Vector2(maxWidth, point.y));
+
+            int intersections = 0;
+            foreach (Edge edge in geometryEdges)
```

Comment:
for loop? (might see perf improvements, but I don't think mobile platforms have bounds)

Reply:
was how it was originally coded... i missed updating this one